auth/genericoauth/provider.go

```diff
@@ -141,9 +142,12 @@ func (GenericTeamProvider) ProviderConstructor(
 	}
 
 	var oauthVerifier verifier.Verifier
-	if genericOAuth.Scope != "" {
+	if genericOAuth.Scope != "" && genericOAuth.Authority = "" {
```

Comment:
@vito Sorry for the late response and many thanks for the input!

Agreed, it does make it counter intuitive. I believe the correct behaviour would be to check and validate both if the user has set both.

Since it looks like I can only use one verifier at a time, I guess this would require me to write a more generic verifier that would accept multiple inputs?